Libraries/LibWeb/DOM/Element.h

```diff
@@ -525,15 +525,26 @@ class Element
     GC::Ptr<CSS::ComputedProperties> m_computed_properties;
     HashMap<FlyString, CSS::StyleProperty> m_custom_properties;
 
-    struct PseudoElement {
+    struct PseudoElement : public JS::Cell {
+        GC_CELL(PseudoElement, JS::Cell);
+        GC_DECLARE_ALLOCATOR(PseudoElement);
+
         GC::Ptr<Layout::NodeWithStyle> layout_node;
         GC::Ptr<CSS::CascadedProperties> cascaded_properties;
         GC::Ptr<CSS::ComputedProperties> computed_properties;
         HashMap<FlyString, CSS::StyleProperty> custom_properties;
+
+    private:
+        virtual void visit_edges(JS::Cell::Visitor&) override;
     };
-    // TODO: CSS::Selector::PseudoElement includes a lot of pseudo-elements that exist in shadow trees,
-    //       and so we don't want to include data for them here.
-    using PseudoElementData = Array<PseudoElement, to_underlying(CSS::PseudoElement::KnownPseudoElementCount)>;
+    // https://drafts.csswg.org/css-view-transitions/#pseudo-element-tree
+    struct PseudoElementTreeNode
```

Comment:
Nothing actually uses the TreeNode abilities of this yet, right? Or am I missing something?

Reply:
Yes, nothing actually uses them in this PR yet.
I wanted to break this out into its own PR since their use in view transitions is tied in with a billion other things and this seemed like a sensible chunk of stuff to PR on its own.